src/Stratis.Bitcoin.Features.Wallet/WalletManager.cs

```diff
@@ -1090,6 +1090,20 @@ public IEnumerable<UnspentOutputReference> GetSpendableTransactionsInWallet(stri
             return res;
         }
 
+        public IEnumerable<TransactionData> GetAllTransactionsInWallet(string walletName)
```

Comment:
That then allows the caller to be more explicit about what they want, rather than relying on the default behaviour of `GetAllTransactions` returning normal account transactions only. We generally would not want the cold staking account transactions to be returned in the wallet RPCs but the watch only account transactions often should be. Cold staking transactions require special care due to their spendability being more complex to determine, so it is safer not to return them.